### PR TITLE
ms-gsl: Update for Conan V2

### PR DESCRIPTION
**Pull request by jwillikers**
Specify library name and version:  **ms-gsl/***

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ms-gsl/all/conanfile.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
-from conans.errors import ConanInvalidConfiguration
 
-required_conan_version = ">=1.43.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import get
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.47.0"
```
Comment (uilianries) [inline recipes/ms-gsl/all/conanfile.py:9]:
```suggestion
required_conan_version = ">=1.50.0"
```
 from conan.tools.build import check_min_cppstd  requires 1.50.0

Comment (jwillikers) [reply recipes/ms-gsl/all/conanfile.py]: Done!

**Review by uilianries — APPROVED** [review]

File: recipes/ms-gsl/all/conanfile.py [hunk 2]
```diff
@@ -43,15 +43,15 @@ def _compilers_minimum_version(self):
         }
 
     def config_options(self):
-        if tools.Version(self.version) >= "3.0.0":
+        if Version(self.version) >= "3.0.0":
             del self.options.on_contract_violation
 
     def configure(self):
```
Comment (toge) [inline recipes/ms-gsl/all/conanfile.py:49]: Please rename `validate`, because compiler version check should be placed in `validate` not `configure`.

**Review by toge — APPROVED** [review]

File: recipes/ms-gsl/all/conanfile.py (hunk 2 above)
Comment (jwillikers) [reply recipes/ms-gsl/all/conanfile.py:49]: I totally missed that. Of course this should use the `validate` method and not `configure`. I've submitted PR #12453 to fix this. Thanks!

**Pull request merged**